### ISSUE TITLE
Don't require "nested" indentation for multiline formulas?

### ISSUE THREAD

**Issue by ThomasSoeiro** [r-lib/lintr#2996]
First, thank you for this very useful package!

Is the following lint intended?

```r
txt <- "
y ~
  var_1 +
  var_2 +
  var_3
"
lintr::lint(
  txt,
  linters = lintr::linters_with_defaults(),
  parse_settings = FALSE
)
#> <text>:4:2: style: [indentation_linter] Indentation should be 4 spaces but is 2 spaces.
#>   var_2 +
#>  ^~~
```

`lintr` currently expects a "nested" indentation:

```r
txt <- "
y ~
  var_1 +
    var_2 +
    var_3
"
lintr::lint(
  txt,
  linters = lintr::linters_with_defaults(),
  parse_settings = FALSE
)
#> ℹ No lints found.
```

I think it is suboptimal because there is no "hierarchy" between the variables.

FWIW, there have been similar discussions for air here: https://github.com/posit-dev/air/issues/444.

**Comment by MichaelChirico** [comment]
See also #2007, #2960, and https://github.com/tidyverse/style/issues/197.

As I see it, 2 indendations is correct:

 - Indentation 1 is for the LHS and RHS of the formula
 - Indentation 2 is how any arithmetic expression is indented:

    ```r
    x +
      y
    if (TRUE) {
      x +
        y
    }
    ```

So I do think it's WAI, though possibly something to be parameterized.

**Comment by ThomasSoeiro** [comment]
> Note: This was changed in air `0.8.0`. Older versions format as lintr expects. Newer versions format formulas consistent with assignment operators rather than with comparison.

Details here: https://github.com/posit-dev/air/blob/main/CHANGELOG.md#080
